internal/util/grpcclient/client.go

```diff
@@ -157,34 +163,70 @@ func (c *ClientBase) connect(ctx context.Context) error {
 		  }
 		}]}`, c.RetryServiceNameConfig, c.MaxAttempts, c.InitialBackoff, c.MaxBackoff, c.BackoffMultiplier)
 
-	conn, err := grpc.DialContext(
-		dialContext,
-		addr,
-		grpc.WithInsecure(),
-		grpc.WithBlock(),
-		grpc.WithDefaultCallOptions(
-			grpc.MaxCallRecvMsgSize(c.ClientMaxRecvSize),
-			grpc.MaxCallSendMsgSize(c.ClientMaxSendSize),
-		),
-		grpc.WithUnaryInterceptor(grpcopentracing.UnaryClientInterceptor(opts...)),
-		grpc.WithStreamInterceptor(grpcopentracing.StreamClientInterceptor(opts...)),
-		grpc.WithDefaultServiceConfig(retryPolicy),
-		grpc.WithKeepaliveParams(keepalive.ClientParameters{
-			Time:                c.KeepAliveTime,
-			Timeout:             c.KeepAliveTimeout,
-			PermitWithoutStream: true,
-		}),
-		grpc.WithConnectParams(grpc.ConnectParams{
-			Backoff: backoff.Config{
-				BaseDelay:  100 * time.Millisecond,
-				Multiplier: 1.6,
-				Jitter:     0.2,
-				MaxDelay:   3 * time.Second,
-			},
-			MinConnectTimeout: c.DialTimeout,
-		}),
-		grpc.WithPerRPCCredentials(&Token{Value: crypto.Base64Encode(util.MemberCredID)}),
-	)
+	var conn *grpc.ClientConn
+	if c.encryption {
+		conn, err = grpc.DialContext(
+			dialContext,
+			addr,
+			//grpc.WithInsecure(),
+			// #nosec G402
+			grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{ServerName: c.role, MinVersion: tls.VersionTLS10})),
```

Comment:
since we do not have  server certificate file now, can we  use  grpc.WithTransportCredentials(insecure.NewCredentials()) instead?